src/components/HeaderWithLanguage/CloneImageDialog.tsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2025-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { Formik, useFormikContext } from "formik";
+import { useCallback, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useNavigate } from "react-router-dom";
+import { DialogOpenChangeDetails } from "@ark-ui/react";
+import {
+  DialogContent,
+  Button,
+  DialogBody,
+  DialogHeader,
+  DialogRoot,
+  DialogTitle,
+  DialogTrigger,
+  Text,
+} from "@ndla/primitives";
+import { styled } from "@ndla/styled-system/jsx";
+import { ImageUploadFormElement } from "../../containers/ImageUploader/components/ImageUploadFormElement";
+import { useMessages } from "../../containers/Messages/MessagesProvider";
+import * as imageApi from "../../modules/image/imageApi";
+import { NdlaErrorPayload } from "../../util/resolveJsonOrRejectWithError";
+import { toEditImage } from "../../util/routeHelpers";
+import { DialogCloseButton } from "../DialogCloseButton";
+import { FormActionsContainer, FormContent } from "../FormikForm";
+import validateFormik from "../formikValidationSchema";
+
+interface Props {
+  loading: boolean;
+  imageId: number;
+}
+
+const StyledFormContent = styled(FormContent, {
+  base: {
+    marginInlineEnd: "large",
+    gap: "small",
+  },
+});
+
+interface FormikValuesType {
+  imageFile: Blob | string | undefined;
+}
+
+const formikRules = { imageFile: { required: true } };
+
+export const CloneImageDialog = ({ loading, imageId }: Props) => {
+  const { t, i18n } = useTranslation();
+  const [open, setOpen] = useState(false);
```

Comment:
Tror denne staten kan fjernes! Dialogen trenger vel ikke være controlled

Reply:
Hva om jeg vil potensielt åpne en warning istedenfor dialogen (sånn som du foreslo) i `onOpenChange`?
Tenkte det måtte gjøres sånn, men om det finnes andre måter så er jeg åpen for det altså.
```typescript
const onOpenChange = useCallback(
  (details: DialogOpenChangeDetails) => {
    if (parentFormIsDirty) {
      createMessage({
        translationKey: "form.mustSaveFirst",
        severity: "danger",
        timeToLive: 0,
      });
    } else {
      setOpen(details.open);
    }
  },
  [createMessage, parentFormIsDirty],
);
```

Reply:
Kommentaren min var relatert til nåværende kode! hvis du trenger å styre om den er open fra et annet sted må den være controlled ja 😄